sql/src/main/java/org/apache/druid/sql/calcite/planner/querygen/DruidQueryGenerator.java

```diff
@@ -133,6 +168,42 @@ private interface Vertex
     SourceDesc unwrapSourceDesc();
   }
 
+  enum JoinSupportTweaks
+  {
+    NONE,
+    LEFT,
+    RIGHT;
+
+    static JoinSupportTweaks analyze(DruidNodeStack stack)
+    {
+      if (stack.size() < 2) {
+        return NONE;
+      }
+      DruidLogicalNode possibleJoin = stack.get(stack.size() - 2);
+      if (!(possibleJoin instanceof DruidJoin)) {
+        return NONE;
+      }
+      if (stack.operandIndexStack.get(stack.size() - 1) == 1) {
+        return RIGHT;
+      } else {
+        return LEFT;
+      }
+    }
```

Comment:
I find this method to be a bit obtuse.  I *think* it's trying to figure out what to put on the right hand side (prioritizing queries on the right?). But, `== 1` in the final if doesn't really carry semantic meaning for me.  It would be nice if you could either adjust the code to be a bit more descriptive in what it's doing or add comments that explain the magic numbers.

Reply:
part of the problem was that a `Join` may have the *same* subtree on both left and right - which mandated the need for the `operandIndex`.

By adding some usefull methods to `DruidNodeStack` ; these magic numbers and substraction have more or less disappeared :)

I don't think `JoinSupportTweaks` will be a long lived `enum` ; I just haven't figured it out yet how many more details I'll need later - I have a feeling that later this will become something like: `VertexOptions` with a set of booleans/etc....